xbmc/platform/darwin/tvos/input/LibInputTouch.mm

```diff
@@ -0,0 +1,647 @@
+
+@implementation TVOSLibInputTouch
+
+#pragma mark - gesture methods
+
+- (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer
+    shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer*)otherGestureRecognizer
+{
+  if ([gestureRecognizer isKindOfClass:[UISwipeGestureRecognizer class]] &&
+      [otherGestureRecognizer isKindOfClass:[UIPanGestureRecognizer class]])
+  {
+    return YES;
+  }
+  if ([gestureRecognizer isKindOfClass:[UIPanGestureRecognizer class]] &&
+      [otherGestureRecognizer isKindOfClass:[UILongPressGestureRecognizer class]])
+  {
+    return YES;
+  }
+  if ([gestureRecognizer isKindOfClass:[UITapGestureRecognizer class]] &&
+      [otherGestureRecognizer isKindOfClass:[UIPanGestureRecognizer class]])
+  {
+    return YES;
+  }
+  return NO;
+}
+
+// called before pressesBegan:withEvent: is called on the gesture recognizer
+// for a new press. return NO to prevent the gesture recognizer from seeing this press
+- (BOOL)gestureRecognizer:(UIGestureRecognizer*)gestureRecognizer shouldReceivePress:(UIPress*)press
+{
+  BOOL handled = YES;
+  switch (press.type)
+  {
+    // single press key, but also detect hold and back to tvos.
+    case UIPressTypeMenu:
+      // menu is special.
+      //  a) if at our home view, should return to atv home screen.
+      //  b) if not, let it pass to us.
+      if (CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() == WINDOW_HOME &&
+          !CServiceBroker::GetGUI()->GetWindowManager().HasVisibleModalDialog() &&
+          !g_application.GetAppPlayer().IsPlaying())
+        handled = NO;
+      break;
+
+    // single press keys
+    case UIPressTypeSelect:
+    case UIPressTypePlayPause:
+      break;
+
+    // auto-repeat keys
+    case UIPressTypeUpArrow:
+    case UIPressTypeDownArrow:
+    case UIPressTypeLeftArrow:
+    case UIPressTypeRightArrow:
+      break;
+
+    default:
+      handled = NO;
+  }
+
+  return handled;
+}
+
+- (void)createSwipeGestureRecognizers
+{
+  for (auto swipeDirection :
+       {UISwipeGestureRecognizerDirectionLeft, UISwipeGestureRecognizerDirectionRight,
+        UISwipeGestureRecognizerDirectionUp, UISwipeGestureRecognizerDirectionDown})
+  {
+    auto swipeRecognizer =
+        [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipe:)];
+    swipeRecognizer.delaysTouchesBegan = NO;
+    swipeRecognizer.direction = swipeDirection;
+    swipeRecognizer.delegate = self;
+    [g_xbmcController.glView addGestureRecognizer:swipeRecognizer];
+  }
+}
+
+- (void)createPanGestureRecognizers
+{
+  // for pan gestures with one finger
+  auto pan = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(handlePan:)];
+  pan.delegate = self;
+  [g_xbmcController.glView addGestureRecognizer:pan];
+}
+
+- (void)createTapGesturecognizers
+{
+  // tap side of siri remote pad
+  for (auto t : {
+         std::make_tuple(UIPressTypeUpArrow, @selector(tapUpArrowPressed:),
```

Comment:
should be possible to use `{}` instead of std::make_tuple(), at least for all params after the first